R/guide-legend.r

```diff
@@ -249,7 +256,18 @@ guide_geom.legend <- function(guide, layers, default_mapping) {
 
     if (length(matched) > 0) {
       # This layer contributes to the legend
-      if (is.na(layer$show.legend) || layer$show.legend) {
+
+      # rename color to colour
+      layer$show.legend <- unlist(rename_aes(as.list(layer$show.legend)))
```

Comment:
When I first tested this, I thought the `rename_aes()` call was not working as intended because _layer$show.legend_ is a named vector and not a list. But it does actually work so I must have been mistaken. I'm fixing this.